src/nvim/ops.c

```diff
@@ -2361,11 +2324,29 @@ static void free_yank_all(void)
  *
  * Return FAIL for failure, OK otherwise.
  */
-int op_yank(oparg_T *oap, int deleting, int mess)
+int op_yank(oparg_T *oap, bool mess) {
```

Comment:
The "enum" seems to actually be `OK`/`FAIL`/`NOTDONE` but `NOTDONE` is only used in one return value in `edit.c` and twice in `eval.c` (just internally in the parsing), I suppose the rest could be refactored to use true/false ( i e every case where `NOTDONE` not possible)